dwl.c

```diff
@@ -597,7 +597,7 @@ focusclient(Client *c, struct wlr_surface *surface, int lift)
 	 * automatically send key events to the appropriate clients.  If surface
 	 * is NULL, this will clear focus.
 	 */
-	if (surface != psurface) {
+	if (surface && (surface != psurface)) {
```

Comment:
I think this needs a corresponding clear_focus as well.  How about `if (!surface)` and then `else if (surface != psurface)`?